quickstarts/04-message-router-workflow/README.md

```diff
@@ -0,0 +1,539 @@
+source .venv/bin/activate
+
+# Install dependencies
+pip install -r requirements.txt
+```
+
+## Configuration
+
+The quickstart includes an OpenAI component configuration in the `components` directory. You have two options to configure your API key:
+
+### Option 1: Using Environment Variables (Recommended)
+
+1. Create a `.env` file in the project root and add your OpenAI API key:
+
+```env
+OPENAI_API_KEY=your_api_key_here
+```
+
+2. When running the examples with Dapr, use the helper script to resolve environment variables:
+
+```bash
+# Get the environment variables from the .env file:
+export $(grep -v '^#' ../../.env | xargs)
+
+# Create a temporary resources folder with resolved environment variables
+temp_resources_folder=$(../resolve_env_templates.py ./components)
+
+# Run your dapr command with the temporary resources
+dapr run --app-id dapr-agent-wf --resources-path $temp_resources_folder -- python sequential_workflow.py
+
+# Clean up when done
+rm -rf $temp_resources_folder
+```
+
+> The temporary resources folder will be automatically deleted when the Dapr sidecar is stopped or when the computer is restarted.
+
+### Option 2: Direct Component Configuration
+
+You can directly update the `key` in [components/openai.yaml](components/openai.yaml):
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: openai
+spec:
+  type: conversation.openai
+  metadata:
+    - name: key
+      value: "YOUR_OPENAI_API_KEY"
+```
+
+Replace `YOUR_OPENAI_API_KEY` with your actual OpenAI API key.
+
+> Many LLM providers are compatible with OpenAI's API (DeepSeek, Google AI, etc.) and can be used with this component by configuring the appropriate parameters. Dapr also has [native support](https://docs.dapr.io/reference/components-reference/supported-conversation/) for other providers like Google AI, Anthropic, Mistral, DeepSeek, etc.
+
+### Additional Components
+
+Make sure Dapr is initialized on your system:
+
+```bash
+dapr init
+```
+
+The quickstart includes other necessary Dapr components in the `components` directory. For example, the workflow state store component:
+
+Look at the `workflowstate.yaml` file in the `components` directory:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: workflowstatestore
+spec:
+  type: state.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+  - name: actorStateStore
+    value: "true"
+```
+
+## Project layout
+
+```text
+04-message-router-workflow/
+├─ components/                 # Dapr components (pubsub, conversation, workflow state)
+├─ app.py                      # Starts WorkflowRuntime + subscribes message routers
+├─ handlers.py                 # @message_router handler (validates + schedules workflow)
+├─ workflow.py                 # workflow & activities (decorated with @llm_activity)
+└─ message_client.py           # publishes a test message to the topic
+```
+
+## Files
```

Comment:
i think it might be nice moving forward for us to not put the code in the readmes and just reference/link the files/components. Thoughts?